core/sts-3.3.md

```diff
@@ -96,28 +100,94 @@ Servers MUST send this key to non-securely connected clients.
 
-## Security
+### User-declared STS policy
+
+Clients may allow users to explicitly define an STS policy for a given host, before any
+interaction with the host. This could help prevent a "Bootstrap MITM vulnerability" as
+discussed in General security considerations.
+
+### Pre-loaded STS policies
+
+As further protection against bootstrap MITM vulnerabilities, clients may choose to
+include a pre-loaded list of known hosts with STS policies. Such lists should be
+compiled on an opt-in basis, by request of IRC network administrators. Hosts should be
+verified to be correctly advertising an STS policy before inclusion.
+
+Clients should consider how their release upgrade cycle compares to server policy expiry
+times when compiling pre-loaded lists. Implementations should be able to provide updates
+to user installed pre-load lists within an appropriate time frame of host policies expiring.
+
+### STS policy deletion or rejection
+
+Clients should allow users to delete cached STS policies on a per-host
+basis, in case a server's policy is accidentally or maliciously injected on
+a secure connection.
+
+Clients may additionally provide the ability to reject STS policies on a
+per-host basis as an additional mitigation.
+
+These features should be made available very carefully in both the user interface
+and security senses. Deleting or rejecting a cache entry for a known STS host should
+be a very deliberate and well-considered act -- it shouldn't be something that users
+get used to doing as a matter of course: e.g., just "clicking through" in order
+to get work done. In other words, these features should not violate the "no immediate
+user recourse" section.
+
+## Server implementation considerations
+
+This section is non-normative.
 
-### STS policy injection
+### Policy expiry time
 
-It is possible for attackers to inject the STS policy into a plaintext
-connection. This causes the client to reconnect over TLS and either fail,
-or connect successfully over TLS but discover that the `sts` cap is not
-present.
+IRCd configurations and network administrators should consider whether expiry times
+represent a constant value into the future, or a fixed expiry time.
 
-### Policy expiration
+Constant values into the future can be achieved by a configured number of seconds being
+sent in the `duration` key on each connection attempt.
+
+Fixed expiry times will involve a dynamic `duration` value being calculated on each
+connection attempt.
+
+Which approach to take will depend on a number of considerations. For example, a server
+may wish their STS Policy to expire at the same time as their domain certificate.
+Alternatively, a server may wish their STS policy to last for as long as possible.
+
+Server implementations should consider employing a value of `duration=0` in their default
+configurations. This will require server administrators to deliberately set an expiry
+according to their specific needs rather than an arbitrary generic value.
```

Comment:
This is meant in e.g. the default example config, and would probably also be commented out. It's just to avoid example configs having arbitrary values that people will blindly choose instead of deciding what suits them.